Barotrauma/BarotraumaShared/SharedSource/Events/Missions/GoToMission.cs

```diff
@@ -4,9 +4,21 @@ namespace Barotrauma
 {
     partial class GoToMission : Mission
     {
+        private readonly bool stateControlsCompletion;
+        private readonly int successState;
+        private readonly int displayAsFailedState;
```

Comment:
I would've still expected this to control whether the mission fails, not whether it _appears to_ fail to clients. The latter seems like it would be a very useful feature (you can control the failure state with scripted events for example), but I don't really see what the current behavior would be useful for.

Reply:
Looking at your earlier comment, I think there might be a small misunderstanding here

> The mission implicitly fails if it's in the fail state as it's not in the success state at the end of the round

This is not the case. Not succeeding is not the same thing as failing: some missions have conditions for failure, such as if the hostages die in a rescue mission, but if you e.g. don't even try to do the objective, you'll neither fail or succeed. At the moment, goto missions can never fail (there's no failure condition).

Reply:
The intended use case is in conjunction with the state controlled success logic to display to clients that the mission has been failed. As you said, GoTo missions can never fail so using it without also using the state controlled success logic wouldn't make much sense.

Reply:
I've updated the logic to only run if the mission's success state is controlled by the state to hopefully make the purpose of failstate clearer as it only makes sense to use with GoTo missions that are controlled by the state.